### ISSUE TITLE
Large documentation pages take a long time to load

### ISSUE THREAD

**Issue by robertknight** [pytorch/pytorch#14686]
## 🐛 Bug

Long pages in the documentation such as the [documentation for the torch.nn module](https://pytorch.org/docs/master/nn.html) take a long time to render and become interactive. On my system in the page is unresponsive for several seconds while opening this page, which makes navigating around the documentation a bit painful.

## To Reproduce

1. Go to the [torch.nn](https://pytorch.org/docs/master/nn.html) docs page in Chrome, Safari or Firefox
1. Try to scroll or interact with the page (eg. to go to docs for a specific class from the table of contents) a few seconds after load

A Chrome trace shows the time is split between layout and scripting. A significant chunk of the scripting time is spent in KaTeX math rendering.

<img width="314" alt="chrome performance summary" src="https://user-images.githubusercontent.com/2458/49345714-ad631a00-f680-11e8-9811-58a0ea0d886b.png">

## Expected behavior

Ideally the page would be small enough that browsers can load it without becoming unresponsive for a significant amount of time.

## Environment

Tested with Chrome 72 and Safari 12.0 on a late 2016 13" MBP.

My knowledge of Sphinx is limited, but perhaps it would make sense to split up the documentation for some of the largest modules into a summary page listing the classes/methods/functions with brief descriptions and a details page with the full docs, eg. using [sphinx.ext.autosummary](http://www.sphinx-doc.org/en/master/usage/extensions/autosummary.html)?

It probably would be possible to optimize the math rendering step without changing the structure, but as long as the page has as many DOM elements as this one, everything is likely to be sluggish.

**Comment by DNGros** [comment]
+1 here. This might not be easy with the way that docs are currently generated, but it would be  nice if docs pages weren't so massive. For example if torch.nn was split up with a different page with each of the submodules on a different page. 

This would both make render times faster, and make it less overwhelming when following a web search result for something like "pytorch loss functions" or "pytorch convolution layer".

Also linking a somewhat duplicate issue in #15760

**Comment by j-haj** [comment]
I have noticed that if I leave the page open in a tab, navigate away for a few minutes and return, the page reloads, which takes a second or two. I would be ok with a slower page load if it only happened once per visit to the page.

**Comment by rgommers** [comment]
This was fixed in the doc restructure in gh-37419. https://pytorch.org/docs/master/nn.html loads fast now. So closing this.
